metaflow/plugins/metadata/local.py

```diff
@@ -112,6 +115,35 @@ def register_metadata(self, run_id, step_name, task_id, metadata):
         }
         self._save_meta(meta_dir, metadict)
 
+    @classmethod
+    def _mutate_user_tags_for_run(
+        cls, flow_id, run_id, tags_to_add=None, tags_to_remove=None
+    ):
+        # get existing tags
+        run = LocalMetadataProvider.get_object("run", "self", {}, None, flow_id, run_id)
+        if not run:
+            raise MetaflowTaggingError(msg="Run not found (%s, %s)" % (flow_id, run_id))
+        existing_user_tags = run["tags"]
+        existing_system_tags = run["system_tags"]
+
+        for tag in tags_to_remove:
+            if tag in existing_system_tags:
+                raise MetaflowTaggingError(
+                    msg="Cannot remove a tag that is an existing system tag (%s)" % tag
+                )
+
+        # make sure no existing system tags get added as a user tag
+        tags_to_add_set = set(tags_to_add) - set(existing_system_tags)
+        # remove tags first, then add
+        next_user_tags_set = (
```

Comment:
So here, removing and adding the same tag would result in a no-op which I guess is what we want but do we warn the user?